File: gateway-android/src/main/kotlin/com/example/gateway/ProcessingTimes.kt
package com.example.gateway

import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response
import okhttp3.HttpUrl.Companion.toHttpUrl

object ProcessingTimes {
    private const val API_URL = "https://egov.uscis.gov/processing-times/api/"
    private val client = OkHttpClient()

    /**
     * Fetch processing time information for a form and office using the
     * public USCIS Processing Times API. Returns a stub if the request
     * fails.
     */
    fun getProcessingTime(formNumber: String, officeCode: String): Map<String, Any> {
        if (formNumber.isBlank() || officeCode.isBlank()) {
            throw IllegalArgumentException("form_number and office_code required")
        }

        val url = "${API_URL}forms/${formNumber}/offices/${officeCode}".toHttpUrl()
        val request = Request.Builder().url(url).build()
        return try {
            client.newCall(request).execute().use { response ->
                ensureSuccess(response)
                val body = response.body?.string() ?: "{}"
                mapOf(
                    "form_number" to formNumber,
                    "office_code" to officeCode,
                    "raw" to body
                )
            }
        } catch (e: Exception) {
            mapOf(
                "form_number" to formNumber,
                "office_code" to officeCode,
                "processing_time" to "N/A",
                "message" to "Processing time API integration pending"
            )
        }
    }

    private fun ensureSuccess(response: Response) {
        if (!response.isSuccessful) {
            throw RuntimeException("Processing Time API failed: ${'$'}{response.code}")
        }
    }
}

File: gateway/processing_times.py
import requests

API_URL = "https://egov.uscis.gov/processing-times/api/"

def get_processing_time(form_number, office_code):
    """Fetch processing time information for a form and office.

    This implementation uses the public USCIS Processing Times API
    which does not require authentication. In the MVP this function
    will likely need to issue multiple requests to list form types and
    field offices. For now it returns a dummy response.
    """
    if not form_number or not office_code:
        raise ValueError("form_number and office_code required")

    # Example request (commented out):
    # resp = requests.get(f"{API_URL}forms/{form_number}/offices/{office_code}")
    # resp.raise_for_status()
    # return resp.json()

    return {
        "form_number": form_number,
        "office_code": office_code,
        "processing_time": "N/A",
        "message": "Processing time API integration pending"
    }
